File: DetectorDeNumerosPrimos/app/src/main/java/com/example/detectordenumerosprimos/MainActivity.java
package com.example.detectordenumerosprimos;

import static operations.Operations.isPrime;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button myButton = null;
    private TextView myTextView = null;
    private EditText myEditText = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        myButton = findViewById(R.id.button);
        myTextView = findViewById(R.id.textView);
        myEditText = findViewById(R.id.editTextText2);

        myButton.setOnClickListener(this);
    }

    public void onClick(View v) {

        String textoB = myButton.getText().toString();

        if (textoB.equals(getString(R.string.text_comprobar))) {
            comprobar();
        } else {
            reboot();

        }
    }

    private void comprobar() {

        try {
            int numero = Integer.parseInt(myEditText.getText().toString());
            if (isPrime(numero)) {
                myTextView.setText(R.string.text_es);
            } else {
                myTextView.setText(R.string.text_noEs);
            }
        } finally {
            myButton.setText(R.string.text_reiniciar);
        }
    }

    public void reboot() {

        myEditText.setText("");
        myTextView.setText(R.string.text_introducir);
        myButton.setText(R.string.text_comprobar);

    }


}

File: LLamarAWalter/app/src/main/java/com/example/llamarawalter/MainActivity.kt
package com.example.llamarawalter

import android.os.Bundle
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val textView: TextView = findViewById(R.id.textView2)
        val imageButton: ImageButton = findViewById(R.id.imageButton3)
        val imageButtonColgar: ImageButton = findViewById(R.id.imageButtonColgar)
        val imageView: ImageView = findViewById(R.id.imageView)
        imageButton.setOnClickListener {
            imageView.setImageResource(R.drawable.walterwhite2)
            textView.setText(R.string.llamando)
            imageButton.visibility = ImageButton.INVISIBLE
            imageButtonColgar.visibility = ImageButton.VISIBLE
        }
        imageButtonColgar.setOnClickListener {
            imageView.setImageResource(R.drawable.walterwhite)
            textView.setText(R.string.llamada_terminada)
            imageButton.visibility = ImageButton.VISIBLE
            imageButtonColgar.visibility = ImageButton.INVISIBLE
        }
    }
}
